lib/Serialization/Deserialization.cpp

```diff
@@ -3323,7 +3323,8 @@ class swift::DeclDeserializer {
                                              None, /*TrailingWhere=*/nullptr);
     declOrOffset = proto;
 
-    proto->setRequiresClass(isClassBounded);
+    ctx.evaluator.cacheOutput(ProtocolRequiresClassRequest{proto},
+                              std::move(isClassBounded));
```

Comment:
The `std::move` isn't technically needed here, because `isClassBounded` is a `bool`, which isn't expensive to copy.

Reply:
That was my first thought, but it doesn't compile just sending in `isClassBounded`. Is it better to copy-construct a temporary?

`ctx.evaluator.cacheOutput(ProtocolRequiresClassRequest{proto}, bool(isClassBounded));`

Reply:
I ran into the same issue recently and used the same workaround, see Deserialization.cpp:

```
      ctx.evaluator.cacheOutput(
          LazyStoragePropertyRequest{var}, std::move(storage));
```

Reply:
Oh, I'm so sorry! I had completely forgotten about the way in which `cacheOutput` is defined. Please go ahead and merge your code as-is; I'll rethink how `cacheOutput` takes its parameter and clean up all of the callers at once.